paper-tooltip.html

```diff
@@ -202,53 +268,69 @@
             }
           }
         },
-
         _showing: {
           type: Boolean,
           value: false
         }
       },
-
       listeners: {
-        'neon-animation-finish': '_onAnimationFinish',
+        'webkitAnimationEnd': '_onAnimationEnd',
       },
-
       /**
        * Returns the target element that this tooltip is anchored to. It is
        * either the element given by the `for` attribute, or the immediate parent
        * of the tooltip.
        */
-      get target () {
+      get target() {
         var parentNode = Polymer.dom(this).parentNode;
         // If the parentNode is a document fragment, then we need to use the host.
         var ownerRoot = Polymer.dom(this).getOwnerRoot();
-
         var target;
         if (this.for) {
           target = Polymer.dom(ownerRoot).querySelector('#' + this.for);
         } else {
           target = parentNode.nodeType == Node.DOCUMENT_FRAGMENT_NODE ?
-              ownerRoot.host : parentNode;
+            ownerRoot.host : parentNode;
         }
-
         return target;
       },
-
-      attached: function() {
+      attached: function () {
         this._findTarget();
       },
-
-      detached: function() {
+      detached: function () {
         if (!this.manualMode)
           this._removeListeners();
       },
-
-      show: function() {
+      /**
+       * Replaces Neon-Animation playAnimation - just calls show and hide.
+       * @deprecated Use show and hide instead.
+       * @param {string} type Either `entry` or `exit`
+       */
+      playAnimation: function (type) {
+        if (type === "entry") {
+          this.show();
+        } else if (type === "exit") {
+          this.hide();
+        }
+      },
+      /**
+       * Cancels the animation and hides the tooltip
+       */
+      cancelAnimation: function () {
+        // Short-cut and cancel all animations and hide
+        this.$.tooltip.classList.remove(this._getAnimationType('entry'));
+        this.$.tooltip.classList.remove(this._getAnimationType('exit'));
+        this.$.tooltip.classList.add('hidden');
```

Comment:
Cancelling animation should not necessarily hide the tooltip; it should just skip the animation

Reply:
Not sure how that links with playAnimation which I thought was it's opposite number. Since the tooltip will most likely be triggered by mouse movement I figured that by calling playAnimation it would of been a programmatic way of opening the tooltip. Thus cancelAnimation would be it's counter and hide the tooltip. Have I misunderstood the reason why playAnimation exists?  By removing the 'entry' and 'exit' the tooltip will disappear anyhow. The hidden just removes from the page completely the tooltip so it doesn't cause page issues.

Your thoughts?

Reply:
I suppose a better question is when cancelAnimation is called. Is the tooltip been asked to complete to fully show itself or completely hide itself? Or does it depend on when it's called? The same is easier with playAnimation where it is set by 'entry' or 'exit'.

Reply:
The previous way it worked is that there would be an entry and exit animation. playAnimation would play this fade in or fade out. cancelAnimation would stop playing the animation and return it to it's pre-animation state.

so if it is in the middle of playing the entry animation, cancel animation would make it hidden. if it is in the middle of playing the exit animation, cancel animation would return it to being visible.